pkg/api/convertertoapi.go

```diff
@@ -640,3 +649,25 @@ func convertVLabsCertificateProfile(vlabs *vlabs.CertificateProfile, api *Certif
 	api.KubeConfigPrivateKey = vlabs.KubeConfigPrivateKey
 	api.SetCAPrivateKey(vlabs.GetCAPrivateKey())
 }
+
+func addDCOSPublicAgentPool(api *Properties) {
+	publicPool := &AgentPoolProfile{}
+	publicPool.Name = api.AgentPoolProfiles[0].Name + "_public"
```

Comment:
@JackQuincy Here's something that we are doing when we add the public pool. When doing the conversion to version-specific, we could look out for the `_public` suffix as a blacklist pattern.